File: Sample/NendSDK_Sample/NativeAdCustomViewController.m
//
//  NativeAdCustomViewController.m
//  NendSDK_Sample
//
//  Copyright © 2016年 F@N Communications. All rights reserved.
//

#import <NendAd/NADNativeClient.h>
#import "NativeAdCustomViewController.h"
#define timerInterval     30.0f

@interface NativeAdCustomViewController () <NADNativeDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *adImageView;
@property (weak, nonatomic) IBOutlet UIImageView *logoImageView;
@property (weak, nonatomic) IBOutlet UILabel *shortTextLabel;
@property (weak, nonatomic) IBOutlet UILabel *longTextLabel;
@property (weak, nonatomic) IBOutlet UILabel *prTextLabel;
@property (weak, nonatomic) IBOutlet UILabel *promotionNameLabel;
@property (weak, nonatomic) IBOutlet UILabel *promotionUrlLabel;
@property (weak, nonatomic) IBOutlet UILabel *actionButtonTextLabel;
@property (weak, nonatomic) IBOutlet UIView *adView;

- (IBAction)enableAutoReload:(id)sender;
- (IBAction)disableAutoReload:(id)sender;

@property (nonatomic, strong) NADNativeClient *client;

@end

@implementation NativeAdCustomViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.adView.layer.borderWidth = 1.0;
    self.adView.layer.borderColor = [UIColor darkGrayColor].CGColor;
    self.actionButtonTextLabel.layer.borderColor = self.actionButtonTextLabel.textColor.CGColor;
    self.actionButtonTextLabel.layer.borderWidth = 1;
    self.actionButtonTextLabel.layer.masksToBounds = YES;
    self.actionButtonTextLabel.layer.cornerRadius = 0;
    self.prTextLabel.baselineAdjustment = UIBaselineAdjustmentAlignCenters;
    
    self.client = [[NADNativeClient alloc] initWithSpotId:@"485504" apiKey:@"30fda4b3386e793a14b27bedb4dcd29f03d638e5"];
    
    __weak typeof(self) weakSelf = self;
    [self.client loadWithCompletionBlock:^(NADNative *ad, NSError *error) {
        if (ad) {
            [weakSelf setUpWithAd:ad];
        }
    }];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

- (void) setUpWithAd:(NADNative *)ad {
    ad.delegate = self;
    self.shortTextLabel.text = ad.shortText;
    self.longTextLabel.text = ad.longText;
    self.prTextLabel.text = [ad prTextForAdvertisingExplicitly:NADNativeAdvertisingExplicitlyPromotion];
    self.promotionNameLabel.text = ad.promotionName;
    self.promotionUrlLabel.text = ad.promotionUrl;
    self.actionButtonTextLabel.text = ad.actionButtonText;
    [ad loadAdImageWithCompletionBlock:^(UIImage *loadAdImage) {
        self.adImageView.image = loadAdImage;
    }];
    [ad loadLogoImageWithCompletionBlock:^(UIImage *loadLogoImage) {
        self.logoImageView.image = loadLogoImage;
    }];
    [ad activateAdView:self.adView withPrLabel:self.prTextLabel];
}

- (IBAction)enableAutoReload:(id)sender {
    __weak typeof(self) weakSelf = self;
    [self.client enableAutoReloadWithInterval:timerInterval completionBlock:^(NADNative *ad, NSError *error) {
        if (ad) {
            [weakSelf setUpWithAd:ad];
        }
    }];
}

- (IBAction)disableAutoReload:(id)sender {
    [self.client disableAutoReload];
}

#pragma mark - NADNativeDelegate

- (void)nadNativeDidClickAd:(NADNative *)ad
{
    NSLog(@"nadNativeDidClickAd: %@", ad);
}

@end

File: Sample/NendSDK_Swift_Sample/NativeAdCustomViewController.swift
//
//  NativeAdCustomViewController.swift
//  NendSDK_Sample
//
//  Copyright © 2016年 F@N Communications. All rights reserved.
//

import UIKit
let interval: NSTimeInterval = 30.0

class NativeAdCustomViewController: UIViewController, NADNativeDelegate {

    @IBOutlet weak var adImageView: UIImageView!
    @IBOutlet weak var logoImageView: UIImageView!
    @IBOutlet weak var shortTextLabel: UILabel!
    @IBOutlet weak var longTextLabel: UILabel!
    @IBOutlet weak var prTextLabel: UILabel!
    @IBOutlet weak var promotionNameLabel: UILabel!
    @IBOutlet weak var promotionUrlLabel: UILabel!
    @IBOutlet weak var actionButtonTextLabel: UILabel!
    @IBOutlet weak var adView: UIView!
    
    private var client: NADNativeClient!
    
    deinit {
        print("NativeAdCustomViewController: deinit")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        NADNativeLogger.setLogLevel(.Debug)
        self.client = NADNativeClient(spotId: "485504", apiKey: "30fda4b3386e793a14b27bedb4dcd29f03d638e5")
        
        weak var weakSelf = self
        self.client.loadWithCompletionBlock() { (ad, error) in
            if let nativeAd = ad {
                weakSelf?.setUpWithAd(nativeAd)
            }
        }
    }
    
    private func setUpWithAd(ad: NADNative) {
        ad.delegate = self
        self.shortTextLabel.text = ad.shortText
        self.shortTextLabel.adjustsFontSizeToFitWidth = true;
        self.longTextLabel.text = ad.longText
        self.longTextLabel.adjustsFontSizeToFitWidth = true;
        self.prTextLabel.text = ad.prTextForAdvertisingExplicitly(.Promotion)
        self.prTextLabel.adjustsFontSizeToFitWidth = true;
        self.promotionNameLabel.text = ad.promotionName
        self.promotionUrlLabel.text = ad.promotionUrl
        self.actionButtonTextLabel.text = ad.actionButtonText
        ad.loadAdImageWithCompletionBlock({(loadAdImage) in
            self.adImageView.image = loadAdImage
        })
        ad.loadLogoImageWithCompletionBlock({(loadLogoImage) in
            self.logoImageView.image = loadLogoImage
        })
        ad.activateAdView(adView, withPrLabel: self.prTextLabel)
        self.adView.layer.borderWidth = 1.0;
        self.adView.layer.borderColor = UIColor.darkGrayColor().CGColor;
    }

    @IBAction func enableAutoReload(sender: AnyObject) {
        weak var weakSelf = self
        self.client.enableAutoReloadWithInterval(interval, completionBlock: { (ad, error) in
            if let nativeAd = ad {
                weakSelf?.setUpWithAd(nativeAd)
            }
        })
    }
    
    @IBAction func disableAutoReload(sender: AnyObject) {
        self.client.disableAutoReload()
    }
    
    // MARK: - NADNativeDelegate
    func nadNativeDidClickAd(ad: NADNative!) {
        print(#function)
    }
}